crates/spl-token-extensions-parser/src/extensions/extension.rs

```diff
@@ -3,7 +3,12 @@ use yellowstone_vixen_parser::{Result, ResultExt};
 
 pub fn decode_extension_ix_type<T: TryFrom<u8>>(ix_data: &[u8]) -> Result<T>
 where T::Error: std::error::Error + Send + Sync + 'static {
-    T::try_from(ix_data[0]).parse_err("Error decoding instruction data for token extension")
+
+    let first_byte: u8 = *ix_data.get(0)
+        .ok_or_else(|| yellowstone_vixen_parser::Error::new("Instruction data for token extension is empty"))?;
```

Comment:
```suggestion
    let first_byte: u8 = *ix_data.get(0)
        .ok_or(yellowstone_vixen_parser::Error::new("Instruction data for token extension is empty"))?;
```

nit: there is no allocations or variable capturing in this error so its ok to use ok_or